Frameworks/CoreLocation/CLLocationManager.mm

```
@@ -411,8 +442,52 @@ - (void)_handleLocationUpdate:(WDGGeoposition*)geoposition {
 }
 
 /**
+ * Handles heading change updates.
+ * @param {WDSCompassReading*} compassReading: updated heading values received from Windows.
+ */
+- (void)_handleHeadingUpdate:(WDSCompassReading*)compassReading {
+    @synchronized(self) {
+        // WDS gives an enum for accuracy, whether it cannot be determined or is reliable/unreliable
+        // As per https://msdn.microsoft.com/en-us/library/windows/hardware/dn642102%28v=vs.85%29.aspx,
+        // High accuracy is given by 0-10 degrees of difference between measured and actual North,
+        // approximate accuracy is between 10 and 25, while unreliable is up to 180 degrees. iOS documentation
+        // states that a negative number is returned when the heading is invalid, so 180 degree error gives -1 and
+        // an unknown accuracy gives -2.
+        CLLocationDirection accuracy = 0.0;
+        switch (compassReading.headingAccuracy) {
+            case WDSMagnetometerAccuracyUnknown:
+                accuracy = -2.0;
+                break;
+            case WDSMagnetometerAccuracyUnreliable:
+                accuracy = -1.0;
+                break;
+            case WDSMagnetometerAccuracyApproximate:
+                accuracy = 25.0;
+                break;
+            case WDSMagnetometerAccuracyHigh:
+                accuracy = 10.0;
+                break;
+            default:
+                accuracy = 0;
```

Is this ever expected? Should we log or fail fast here?